Automattic/wp-calypso
# Hosting Config: Add key icon to material-design-icons package

#### Proposed Changes

The credentials card in the `/hosting-config` page currently uses a cloud icon. While working on [the new upsell banner](https://github.com/Automattic/wp-calypso/pull/71750) it was pointed out that the cloud icon on the credential card is clashing with the cloud icon used in the feature list. The same icon is used to represent different concepts.

Since a cloud isn't necessarily the best way to represent the credential card anyway, this PR changes the icon to a key. To me this better invokes key management.

![CleanShot 2023-01-10 at 14 34 56@2x](https://user-images.githubusercontent.com/1500769/211442115-6f5ea16b-a880-48fc-ba6b-211b50553122.png)

![CleanShot 2023-01-10 at 14 36 29@2x](https://user-images.githubusercontent.com/1500769/211442126-0d35d588-7cc3-46ff-b5eb-209bbac3a221.png)

I had to add the key icon to the `@automattic/material-design-icons` package. There are instructions in the README for this.

A note about the file name used for the new SVG file: ID used to look up the icon inside the sprite map includes
the icon size. Since we eventually want the icon to be `size=32` that unfortunately means `32` needs to be in the ID, which means we need "32px" in the SVG file name—even though it's still a 24px SVG under the hood.

This is how the cloud icon we currently use works. It's not ideal but I wasn't going to rework how the `material-design-icons` package works.

Original icon comes from:
https://fonts.google.com/icons?selected=Material+Icons&icon.query=key&icon.style=Outlined&icon.category=communication

#### Testing Instructions



* Head to `/hosting-config` and see that the icon has changed to a key
* Look at the diff with some tool that can show where within a line changes have been made. There shouldn't be any changes to the existing icons in the sprite.
* Find some other places in Calypso where material icons are used and confirm that this PR hasn't broken them.

#### Pre-merge Checklist
